src/test/java/org/mitre/inferno/EndpointTest.java

```diff
@@ -0,0 +1,182 @@
+package org.mitre.inferno;
+
+import org.mitre.inferno.App;
+import org.mitre.inferno.Validator;
+import org.mitre.inferno.rest.IgResponse;
+
+import java.io.IOException;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.BufferedReader;
+import java.util.List;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonParser;
+import org.junit.jupiter.api.BeforeAll;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import org.junit.jupiter.api.Test;
+import io.restassured.RestAssured;
+import io.restassured.response.Response;
+import io.restassured.response.ResponseBody;
+import io.restassured.specification.RequestSpecification;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Represents a tester for api endpoints.
+ */
+public class EndpointTest {
+    private Logger logger = LoggerFactory.getLogger(App.class);
+    private static Validator validator;
+
+    @BeforeAll
+    public static void setup() throws Exception {
+        // get validation results directly from a validator instance
```

Comment:
I'd recommend getting rid of all these comments. Comments should be used only when absolutely necessary. If it isn't clear what a segment of code does, extract that code into a method whose name says what it does.

```java
/* Bad: Explain code with a comment */
// start the wrapper app
String [] args = {};
App.main(args);

/* Better: extract the code into a method which says what it does */
startWrapper();

private void startWrapper() {
  String [] args = {};
  App.main(args);
}

/* Best: have code that just makes sense */
String [] args = {};
App.main(args);
```